app.rb

```diff
@@ -0,0 +1,119 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    i = 0
+    new_array = self
+    new_array = to_a if is_a?(Range) || is_a?(Hash)
+    while i < new_array.length
+      yield new_array[i]
+      i += 1
+    end
+    new_array
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    i = 0
+    new_array = self
+    new_array = to_a if is_a?(Range) || is_a?(Hash)
+    while i < new_array.length
+      yield new_array[i], i
+      i += 1
+    end
+    new_array
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    new_numbers = []
+    my_each { |i| new_numbers.push(i) if yield(i) }
+    new_numbers
+  end
+
+  def my_all?(arg = nil)
+    return false if arg.nil? && !block_given? && !my_each { |j| j }
+
+    unless arg.nil?
+      return false unless my_each { |j| j.class == arg || arg.match(j) }
+    end
+    return false if block_given? && !my_each { |j| yield(j) }
+
+    true
+  end
+
+  def my_any?(arg = nil)
+    return true unless arg.nil? || !my_each { |j| j.class == arg }
+    return true unless block_given? || !my_each { |j| j }
+    return true if block_given? && my_each { |j| yield(j) }
+
+    true
+  end
+
+  def my_none?(arg = nil)
+    !my_any?(arg)
+  end
+
+  def my_count(counted = nil)
+    new_array = is_a?(Range) ? to_a : self
+    return new_array.length if counted.nil? && !block_given?
+
+    counting = 0
+    if block_given?
+      new_array.my_each { |j| counting += 1 if yield j }
+    end
+    unless counted.nil?
+      new_array.my_each { |j| counting += 1 if j == counted }
+    end
+    counting
+  end
```

Comment:
# my_none?
1. [ ]  Returns true if the block never returns true for all elements
### Test 👉 `array.my_none?(&true_block)`
|Expected|Got|
|:---:|:---:|
|`true`|`false`|
#
2. [ ] When no block or argument is given returns true only if none of the collection members is true
### Test 👉 `false_array.my_none?`
|Expected|Got|
|:---:|:---:|
|`true`|`false`|
#
3. [ ] When a Regex is passed as an argument returns true if none of the collection matches the Regex
### Test 👉 `words.my_none?(/z/)`
|Expected|Got|
|:---:|:---:|
|`true`|`false`|
#
4. [ ]  When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
### Test 👉 `words.my_none?(5)`
|Expected|Got|
|:---:|:---:|
|`true`|`false`|
#